// Build command:
// python -m grpc_tools.protoc -I ../ --python_out=./ --grpc_python_out=./ ../gpio.proto

syntax = "proto3";
package GpioController;

enum GpioState {
    Off = 0;
    On = 1;
}

message SetGpioRequest {
    uint32 GpioNumber = 1;
    GpioState State = 2;
}

message GetGpioRequest {
    uint32 GpioNumber = 1;
}

message GetGpioResponse {
    uint32 GpioNumber = 1;
    GpioState State = 2;
}

message SetGpioResponse {
}

message BlinkGpioRequest {
    uint32 GpioNumber = 1;
    double Length = 2;
}

message BlinkGpioResponse {
    uint32 GpioNumber = 1;
    double Length = 2;
}

service Gpio {
    rpc GetGpio (GetGpioRequest) returns (GetGpioResponse) {}
    rpc SetGpio (SetGpioRequest) returns (SetGpioResponse) {}
    rpc BlinkGpio (BlinkGpioRequest) returns (BlinkGpioResponse) {}
}
